code/modules/reagents/chemistry/reagents/food_reagents.dm

```diff
@@ -101,6 +101,7 @@
 			O.loc.visible_message("<span class='warning'>[O] rapidly fries as it's splashed with hot oil! Somehow.</span>")
 			var/obj/item/reagent_containers/food/snacks/deepfryholder/F = new(O.drop_location(), O)
 			F.fry(volume)
+			F.reagents.add_reagent("cooking_oil", reac_volume)
```

Comment:
Just curious, why isn't this handled by the fry proc?

Reply:
There are two answers to that, the real answer is probably that deep fryers were the only user of the fry proc until recently, and they transfer reagents on process(). The optimistic answer is because deep frying works without cooking oil, and whoever coded the deep friers wanted to take that into account. Regardless, we can safely add cooking oil here as this is a cooking oil proc.